internal/include/utils.h

```diff
@@ -20,6 +20,18 @@ static __always_inline bool bpf_memcmp(char *s1, char *s2, s32 size)
     return true;
 }
 
+// assumes s2 is all lowercase
+static __always_inline int bpf_memicmp(const char *s1, const char *s2, s32 size) {
```

Comment:
I added a helper we have used before in Beyla which can do case insensitive compares for the header fields. e.g. Traceparent, TRACEPARENT, TraceParent, traceparent all match.